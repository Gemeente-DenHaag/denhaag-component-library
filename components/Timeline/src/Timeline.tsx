import React from 'react';
import { Stepper as MaterialStepper } from '@material-ui/core';
import BaseProps from '@gemeente-denhaag/baseprops';
import clsx from 'clsx';
import './styles/_mui-override.scss';

export interface TimelineProps extends Omit<BaseProps, 'classes'> {
  /**
   * Set the active step (zero based index). Set to -1 to disable all the steps.
   */
  activeStep?: number;
}

/**
 * The `Timeline` displays a progress through a sequence by breaking it up into
 * multiple logical and numbered steps. Use the timeline when a series of
 * information needs to be ordered by time (ascending or descending). It
 * represents the steps in a process in a chronological order. This process is
 * part of, for example, a request that the user has made.
 *
 * @param props The properties of a Timeline component.
 * @constructor Constructs an instance of Timeline.
 */
export const Timeline: React.FC<TimelineProps> = ({
  activeStep,
  children,
  id = undefined,
  className,
}: TimelineProps) => {
  const timelineClassName = clsx('denhaag-timeline', className);

  return (
    <MaterialStepper
      activeStep={activeStep}
      connector={<React.Fragment />}
      id={id}
      nonLinear
      orientation="vertical"
      className={timelineClassName}
    >
      {children}
    </MaterialStepper>
  );
};

export default Timeline;
