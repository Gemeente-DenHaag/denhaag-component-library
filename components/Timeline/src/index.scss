@import "./styles/step-content";
@import "./styles/step-icon";
@import "./styles/step-label";
@import "./styles/step";
@import "./styles/timeline";
