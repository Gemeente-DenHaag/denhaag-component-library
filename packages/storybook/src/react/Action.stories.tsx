import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { ActionSingle, ActionSingleProps } from '@gemeente-denhaag/action';
import { FormatDateLabels } from '@gemeente-denhaag/utils';

import readme from '../../../../components/Action/README.md';

const labels: FormatDateLabels = {
  today: 'vandaag',
  yesterday: 'gisteren',
  before: 'vóór',
  approachingDeadline: (daysDifference: number) => {
    if (daysDifference === 1) {
      return `nog ${daysDifference} dag`;
    }
    return `nog ${daysDifference} dagen`;
  },
};

const exampleArgs: ActionSingleProps = {
  children: <strong>Taak</strong>,
  link: '#example',
  labels,
};

const meta = {
  id: 'react-action',
  title: 'React/Actions/Action',
  component: ActionSingle,
  args: exampleArgs,
  tags: ['autodocs'],
  parameters: {
    chromatic: { viewports: [1768, 1280, 768, 360] },
    docs: {
      description: {
        component: readme,
      },
    },
  },
} as Meta<typeof ActionSingle>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};

export const Hover: Story = {
  args: { ...Default.args, className: 'denhaag-action--hover' },
};

export const Focus: Story = {
  args: { ...Default.args, className: 'denhaag-action--focus' },
};

export const WithDate: Story = {
  args: { ...Default.args, dateTime: '2023-09-28T19:47:36.593Z' },
};

export const WithDateRelative: Story = {
  args: { ...Default.args, dateTime: '2023-10-02T19:47:36.593Z', now: '2023-09-28T19:47:36.593Z', relativeDate: true },
};

export const WithDateWarning: Story = {
  args: { ...Default.args, dateTime: '2023-09-30T19:47:36.593Z', now: '2023-09-28T19:47:36.593Z', relativeDate: true },
};

// ### Multiple - hover

// <Canvas>
//   <Story name="Multiple - hover">
//     {() => (
//       <Action
//         actions={
//           <ButtonLink href="#example" className="utrecht-button-link--hover">
//             Actie
//           </ButtonLink>
//         }
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>

// ### Multiple - focus

// <Canvas>
//   <Story name="Multiple - focus">
//     {() => (
//       <Action
//         actions={
//           <ButtonLink href="#example" className="utrecht-button-link--focus">
//             Actie
//           </ButtonLink>
//         }
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>

// ### Multiple - with date

// <Canvas>
//   <Story name="Multiple - with date">
//     {() => (
//       <Action
//         dateTime={"2023-09-28T19:47:36.593Z"}
//         now={"2023-09-28T19:47:36.593Z"}
//         labels={labels}
//         actions={<ButtonLink href="#example">Actie</ButtonLink>}
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>

// ### Multiple - with date - relative

// <Canvas>
//   <Story name="Multiple - with date - relative">
//     {() => (
//       <Action
//         relativeDate
//         dateTime={"2023-10-02T19:47:36.593Z"}
//         now={"2023-09-28T19:47:36.593Z"}
//         labels={labels}
//         actions={<ButtonLink href="#example">Actie</ButtonLink>}
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>

// ### Multiple - with warning date - relative

// <Canvas>
//   <Story name="Multiple - with warning date - relative">
//     {() => (
//       <Action
//         relativeDate
//         dateTime={"2023-09-30T19:47:36.593Z"}
//         now={"2023-09-28T19:47:36.593Z"}
//         labels={labels}
//         actions={<ButtonLink href="#example">Actie</ButtonLink>}
//       >
//         <strong>Taak</strong>
//       </Action>
//     )}
//   </Story>
// </Canvas>
