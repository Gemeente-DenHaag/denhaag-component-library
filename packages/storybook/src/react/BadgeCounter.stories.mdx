import { ArgsTable, Canvas, Description, Meta, Story } from "@storybook/blocks";

import { BadgeCounter } from "@gemeente-denhaag/components-react";
import pkg from "../../../../components/Button/package.json";
import readme from "../../../../components/Button/README.md";

<Meta
  title="React/Data Display/Badge Counter"
  component={BadgeCounter}
  parameters={{
    componentSubtitle: `${pkg.name} - ${pkg.version}`,
    docs: { source: { type: "dynamic" } },
    status: {
      type: "ALPHA",
    },
  }}
/>

<Description markdown={readme} />

## Stories

### Default

<Canvas>
  <Story name="Default">{(args) => <BadgeCounter {...args}>13</BadgeCounter>}</Story>
</Canvas>

<ArgsTable of={BadgeCounter} />
