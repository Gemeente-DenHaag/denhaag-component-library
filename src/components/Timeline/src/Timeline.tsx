import React from 'react';
import MaterialStepper from '@material-ui/core/Stepper';
import BaseProps from '@gemeente-denhaag/baseprops';

import { timelineClasses } from './bem-mapping';

export interface TimelineProps extends BaseProps {
  /**
   * Set the active step (zero based index). Set to -1 to disable all the steps.
   */
  activeStep?: number;
}

/**
 * A Stepper conveys progress through numbered steps.
 * @param props The properties of a Timeline component.
 * @constructor Constructs an instance of Timeline.
 */
export const Timeline: React.FC<TimelineProps> = (props: TimelineProps) => {
  return (
    <MaterialStepper
      nonLinear
      orientation="vertical"
      connector={<React.Fragment />}
      classes={timelineClasses}
      {...props}
    >
      {props.children}
    </MaterialStepper>
  );
};

export default Timeline;
