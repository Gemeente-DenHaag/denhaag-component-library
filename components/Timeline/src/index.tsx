import Timeline from './Timeline';

import './styles/mui-override.scss';
import './styles/step-content.scss';
import './styles/step-icon.scss';
import './styles/step-label.scss';
import './styles/step.scss';
import './styles/timeline.scss';

export default Timeline;
export * from './Step';
export * from './StepContent';
export * from './StepIcon';
export * from './StepLabel';
export * from './Timeline';
