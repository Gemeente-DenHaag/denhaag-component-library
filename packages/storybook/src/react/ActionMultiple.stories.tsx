import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { ActionMulti } from '@gemeente-denhaag/action';
import { ButtonLink } from '@utrecht/component-library-react';

import readme from '../../../../components/Action/README.md';

const labels = {
  today: 'vandaag',
  yesterday: 'gisteren',
  before: 'vóór',
  approachingDeadline: (daysDifference: number) => {
    if (daysDifference === 1) {
      return `nog ${daysDifference} dag`;
    }
    return `nog ${daysDifference} dagen`;
  },
};

const exampleArgs = {
  labels,
  children: <strong>Taak</strong>,
  actions: <ButtonLink href="#example">Actie</ButtonLink>,
};

const meta = {
  id: 'react-actions-action-multiple',
  title: 'React/Actions/Action Multiple',
  component: ActionMulti,
  args: exampleArgs,
  tags: ['autodocs'],
  parameters: {
    chromatic: { viewports: [1768, 1280, 768, 360] },
    docs: {
      description: {
        component: readme,
      },
    },
  },
} as Meta<typeof ActionMulti>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};

export const Hover: Story = {
  args: {
    ...Default.args,
    actions: (
      <ButtonLink href="#example" className="utrecht-button-link--hover">
        Actie
      </ButtonLink>
    ),
  },
};

export const Focus: Story = {
  args: {
    ...Default.args,
    actions: (
      <ButtonLink href="#example" className="utrecht-button-link--focus">
        Actie
      </ButtonLink>
    ),
  },
};

export const WithDate: Story = {
  args: { ...Default.args, dateTime: '2023-09-28T19:47:36.593Z' },
};

export const WithDateRelative: Story = {
  args: { ...Default.args, dateTime: '2023-10-02T19:47:36.593Z', now: '2023-09-28T19:47:36.593Z', relativeDate: true },
};

export const WithDateWarning: Story = {
  args: { ...Default.args, dateTime: '2023-09-30T19:47:36.593Z', now: '2023-09-28T19:47:36.593Z', relativeDate: true },
};
