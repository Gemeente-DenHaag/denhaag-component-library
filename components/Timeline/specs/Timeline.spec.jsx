import * as React from 'react';
import Timeline, { Step } from '../src';

describe('Timeline tests', () => {
  it('completed step can show description', () => {
    cy.mount(
      <Timeline activeStep={1}>
        <Step label="Label tekst" description="Description" completed={true} />
      </Timeline>,
    );

    cy.get('.denhaag-timeline__step').should('not.contain', 'Description');
    cy.get('.denhaag-timeline__step-label').click();
    cy.get('.denhaag-timeline__step').should('contain', 'Description');
  });

  it('not active step cannot show description', () => {
    cy.mount(
      <Timeline activeStep={-1}>
        <Step label="Label tekst" description="Description" completed={false} />
      </Timeline>,
    );

    cy.get('.denhaag-timeline__step').should('not.contain', 'Description');
    cy.get('.denhaag-timeline__step-label').click();
    cy.get('.denhaag-timeline__step').should('not.contain', 'Description');
  });

  it('active step cannot hide description', () => {
    cy.mount(
      <Timeline activeStep={0}>
        <Step label="Label tekst" description="Description" completed={false} />
      </Timeline>,
    );

    cy.get('.denhaag-timeline__step').should('contain', 'Description');
    cy.get('.denhaag-timeline__step-label').click();
    cy.get('.denhaag-timeline__step').should('contain', 'Description');
  });

  it('a11y and snapshots', () => {
    const steps = [
      {
        label: 'First Step in a Timeline!',
        description: 'This is the first Step component of many in a Timeline component.',
      },
      {
        label: 'Second Step in a Timeline!',
      },
      {
        label: 'Third Step in a Timeline!',
        disabled: true,
      },
      {
        label: 'Fourth Step in a Timeline!',
        description: 'This is the fourth Step component of many in a Timeline component.',
      },
    ];

    cy.snapshots(
      <div>
        {[0, 1, 3, 4].map((activeStep) => (
          <Timeline activeStep={activeStep} key={activeStep}>
            {steps.map(({ label, ...props }, index) => (
              <Step key={label} label={label} {...props} completed={activeStep > index && !props.disabled} />
            ))}
          </Timeline>
        ))}
      </div>,
      {},
      [
        {
          selector: '.denhaag-timeline__step',
          states: {
            focus: 'denhaag-timeline__step--focus',
          },
        },
        {
          selector: '.denhaag-timeline__step-label--completed',
          states: {
            hover: 'denhaag-timeline__step-label--completed--hover',
          },
        },
      ],
      [
        { id: 'color-contrast', issue: 412 },
        { id: 'aria-allowed-attr', issue: 731 },
      ],
      'Timeline',
    );
  });
});
