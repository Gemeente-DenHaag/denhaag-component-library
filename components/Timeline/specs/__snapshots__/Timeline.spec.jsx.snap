exports[`Timeline tests > a11y and snapshots #0`] = `
<div id="wrapper" style="display: inline-block;">
  <div style="padding: 10px;">
    <div>
      <div
        class="MuiPaper-root MuiStepper-root denhaag-timeline MuiStepper-vertical MuiPaper-elevation0"
      >
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical"
          aria-expanded="false"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--active"
          >
            <svg
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorAction denhaag-timeline__step-icon--active"
              focusable="false"
              viewBox="0 0 20 20"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <circle
                cx="10"
                cy="10"
                r="10"
                fill="currentColor"
                stroke="none"
                stroke-width="0"
              ></circle>
              <text
                x="10"
                y="14"
                text-anchor="middle"
                class="denhaag-timeline__step-icon-text denhaag-timeline__step-icon-text--active"
              >
                1
              </text>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              First Step in a Timeline!
            </p>
          </div>
          <div
            class="MuiStepContent-root denhaag-timeline__step-content"
            icon="1"
          >
            <div
              class="MuiCollapse-root MuiStepContent-transition MuiCollapse-entered"
              style="min-height: 0px;"
            >
              <div class="MuiCollapse-wrapper">
                <div class="MuiCollapse-wrapperInner">
                  This is the first Step component of many in a Timeline
                  component.
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical"
          aria-expanded="false"
          tabindex="0"
        >
          <div class="denhaag-timeline__step-label">
            <svg
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorPrimary denhaag-timeline__step-icon--inactive"
              focusable="false"
              viewBox="0 0 20 20"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <circle
                cx="10"
                cy="10"
                r="9.25"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
              ></circle>
              <text
                x="10"
                y="14"
                text-anchor="middle"
                class="denhaag-timeline__step-icon-text"
              >
                2
              </text>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              Second Step in a Timeline!
            </p>
          </div>
        </div>
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical"
          aria-expanded="false"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--disabled"
          >
            <svg
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorDisabled denhaag-timeline__step-icon--disabled"
              focusable="false"
              viewBox="0 0 20 20"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <circle
                cx="10"
                cy="10"
                r="9.25"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
              ></circle>
              <text
                x="10"
                y="14"
                text-anchor="middle"
                class="denhaag-timeline__step-icon-text"
              >
                3
              </text>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              Third Step in a Timeline!
            </p>
          </div>
        </div>
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical"
          aria-expanded="false"
          tabindex="0"
        >
          <div class="denhaag-timeline__step-label">
            <svg
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorPrimary denhaag-timeline__step-icon--inactive"
              focusable="false"
              viewBox="0 0 20 20"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <circle
                cx="10"
                cy="10"
                r="9.25"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
              ></circle>
              <text
                x="10"
                y="14"
                text-anchor="middle"
                class="denhaag-timeline__step-icon-text"
              >
                4
              </text>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              Fourth Step in a Timeline!
            </p>
          </div>
          <div
            class="MuiStepContent-root denhaag-timeline__step-content MuiStepContent-last"
            icon="4"
          ></div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiStepper-root denhaag-timeline MuiStepper-vertical MuiPaper-elevation0"
      >
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical MuiStep-completed denhaag-timeline__step--completed"
          aria-expanded="false"
          role="button"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--completed"
          >
            <svg
              width="1em"
              height="1em"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorSecondary denhaag-timeline__step-icon--completed"
              focusable="false"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <path
                d="M22 11.08V12a10 10 0 11-5.93-9.14"
                stroke="currentColor"
                fill="none"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              ></path>
              <path
                d="M22 4L12 14.01l-3-3"
                stroke="currentColor"
                fill="none"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              ></path>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              First Step in a Timeline!<svg
                width="1em"
                height="1em"
                viewBox="0 0 24 24"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
                class="MuiSvgIcon-root denhaag-timeline__step-collapse-icon"
                focusable="false"
                aria-hidden="true"
                shape-rendering="auto"
              >
                <path
                  d="M5.293 8.293a1 1 0 011.414 0L12 13.586l5.293-5.293a1 1 0 111.414 1.414l-6 6a1 1 0 01-1.414 0l-6-6a1 1 0 010-1.414z"
                  fill="currentColor"
                ></path>
              </svg>
            </p>
          </div>
          <div
            class="MuiStepContent-root denhaag-timeline__step-content"
            icon="1"
          ></div>
        </div>
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical"
          aria-expanded="false"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--active"
          >
            <svg
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorAction denhaag-timeline__step-icon--active"
              focusable="false"
              viewBox="0 0 20 20"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <circle
                cx="10"
                cy="10"
                r="10"
                fill="currentColor"
                stroke="none"
                stroke-width="0"
              ></circle>
              <text
                x="10"
                y="14"
                text-anchor="middle"
                class="denhaag-timeline__step-icon-text denhaag-timeline__step-icon-text--active"
              >
                2
              </text>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              Second Step in a Timeline!
            </p>
          </div>
        </div>
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical"
          aria-expanded="false"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--disabled"
          >
            <svg
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorDisabled denhaag-timeline__step-icon--disabled"
              focusable="false"
              viewBox="0 0 20 20"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <circle
                cx="10"
                cy="10"
                r="9.25"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
              ></circle>
              <text
                x="10"
                y="14"
                text-anchor="middle"
                class="denhaag-timeline__step-icon-text"
              >
                3
              </text>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              Third Step in a Timeline!
            </p>
          </div>
        </div>
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical"
          aria-expanded="false"
          tabindex="0"
        >
          <div class="denhaag-timeline__step-label">
            <svg
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorPrimary denhaag-timeline__step-icon--inactive"
              focusable="false"
              viewBox="0 0 20 20"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <circle
                cx="10"
                cy="10"
                r="9.25"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
              ></circle>
              <text
                x="10"
                y="14"
                text-anchor="middle"
                class="denhaag-timeline__step-icon-text"
              >
                4
              </text>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              Fourth Step in a Timeline!
            </p>
          </div>
          <div
            class="MuiStepContent-root denhaag-timeline__step-content MuiStepContent-last"
            icon="4"
          ></div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiStepper-root denhaag-timeline MuiStepper-vertical MuiPaper-elevation0"
      >
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical MuiStep-completed denhaag-timeline__step--completed"
          aria-expanded="false"
          role="button"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--completed"
          >
            <svg
              width="1em"
              height="1em"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorSecondary denhaag-timeline__step-icon--completed"
              focusable="false"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <path
                d="M22 11.08V12a10 10 0 11-5.93-9.14"
                stroke="currentColor"
                fill="none"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              ></path>
              <path
                d="M22 4L12 14.01l-3-3"
                stroke="currentColor"
                fill="none"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              ></path>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              First Step in a Timeline!<svg
                width="1em"
                height="1em"
                viewBox="0 0 24 24"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
                class="MuiSvgIcon-root denhaag-timeline__step-collapse-icon"
                focusable="false"
                aria-hidden="true"
                shape-rendering="auto"
              >
                <path
                  d="M5.293 8.293a1 1 0 011.414 0L12 13.586l5.293-5.293a1 1 0 111.414 1.414l-6 6a1 1 0 01-1.414 0l-6-6a1 1 0 010-1.414z"
                  fill="currentColor"
                ></path>
              </svg>
            </p>
          </div>
          <div
            class="MuiStepContent-root denhaag-timeline__step-content"
            icon="1"
          ></div>
        </div>
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical MuiStep-completed denhaag-timeline__step--completed"
          aria-expanded="false"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--completed"
          >
            <svg
              width="1em"
              height="1em"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorSecondary denhaag-timeline__step-icon--completed"
              focusable="false"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <path
                d="M22 11.08V12a10 10 0 11-5.93-9.14"
                stroke="currentColor"
                fill="none"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              ></path>
              <path
                d="M22 4L12 14.01l-3-3"
                stroke="currentColor"
                fill="none"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              ></path>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              Second Step in a Timeline!
            </p>
          </div>
        </div>
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical"
          aria-expanded="false"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--disabled"
          >
            <svg
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorDisabled denhaag-timeline__step-icon--disabled"
              focusable="false"
              viewBox="0 0 20 20"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <circle
                cx="10"
                cy="10"
                r="9.25"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
              ></circle>
              <text
                x="10"
                y="14"
                text-anchor="middle"
                class="denhaag-timeline__step-icon-text"
              >
                3
              </text>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              Third Step in a Timeline!
            </p>
          </div>
        </div>
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical"
          aria-expanded="false"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--active"
          >
            <svg
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorAction denhaag-timeline__step-icon--active"
              focusable="false"
              viewBox="0 0 20 20"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <circle
                cx="10"
                cy="10"
                r="10"
                fill="currentColor"
                stroke="none"
                stroke-width="0"
              ></circle>
              <text
                x="10"
                y="14"
                text-anchor="middle"
                class="denhaag-timeline__step-icon-text denhaag-timeline__step-icon-text--active"
              >
                4
              </text>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              Fourth Step in a Timeline!
            </p>
          </div>
          <div
            class="MuiStepContent-root denhaag-timeline__step-content MuiStepContent-last"
            icon="4"
          >
            <div
              class="MuiCollapse-root MuiStepContent-transition MuiCollapse-entered"
              style="min-height: 0px;"
            >
              <div class="MuiCollapse-wrapper">
                <div class="MuiCollapse-wrapperInner">
                  This is the fourth Step component of many in a Timeline
                  component.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiStepper-root denhaag-timeline MuiStepper-vertical MuiPaper-elevation0"
      >
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical MuiStep-completed denhaag-timeline__step--completed"
          aria-expanded="false"
          role="button"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--completed"
          >
            <svg
              width="1em"
              height="1em"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorSecondary denhaag-timeline__step-icon--completed"
              focusable="false"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <path
                d="M22 11.08V12a10 10 0 11-5.93-9.14"
                stroke="currentColor"
                fill="none"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              ></path>
              <path
                d="M22 4L12 14.01l-3-3"
                stroke="currentColor"
                fill="none"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              ></path>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              First Step in a Timeline!<svg
                width="1em"
                height="1em"
                viewBox="0 0 24 24"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
                class="MuiSvgIcon-root denhaag-timeline__step-collapse-icon"
                focusable="false"
                aria-hidden="true"
                shape-rendering="auto"
              >
                <path
                  d="M5.293 8.293a1 1 0 011.414 0L12 13.586l5.293-5.293a1 1 0 111.414 1.414l-6 6a1 1 0 01-1.414 0l-6-6a1 1 0 010-1.414z"
                  fill="currentColor"
                ></path>
              </svg>
            </p>
          </div>
          <div
            class="MuiStepContent-root denhaag-timeline__step-content"
            icon="1"
          ></div>
        </div>
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical MuiStep-completed denhaag-timeline__step--completed"
          aria-expanded="false"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--completed"
          >
            <svg
              width="1em"
              height="1em"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorSecondary denhaag-timeline__step-icon--completed"
              focusable="false"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <path
                d="M22 11.08V12a10 10 0 11-5.93-9.14"
                stroke="currentColor"
                fill="none"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              ></path>
              <path
                d="M22 4L12 14.01l-3-3"
                stroke="currentColor"
                fill="none"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              ></path>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              Second Step in a Timeline!
            </p>
          </div>
        </div>
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical"
          aria-expanded="false"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--disabled"
          >
            <svg
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorDisabled denhaag-timeline__step-icon--disabled"
              focusable="false"
              viewBox="0 0 20 20"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <circle
                cx="10"
                cy="10"
                r="9.25"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
              ></circle>
              <text
                x="10"
                y="14"
                text-anchor="middle"
                class="denhaag-timeline__step-icon-text"
              >
                3
              </text>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              Third Step in a Timeline!
            </p>
          </div>
        </div>
        <div
          class="MuiStep-root denhaag-timeline__step MuiStep-vertical MuiStep-completed denhaag-timeline__step--completed"
          aria-expanded="false"
          role="button"
          tabindex="0"
        >
          <div
            class="denhaag-timeline__step-label denhaag-timeline__step-label--completed"
          >
            <svg
              width="1em"
              height="1em"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
              class="MuiSvgIcon-root denhaag-timeline__step-icon MuiSvgIcon-colorSecondary denhaag-timeline__step-icon--completed"
              focusable="false"
              aria-hidden="true"
              shape-rendering="auto"
            >
              <path
                d="M22 11.08V12a10 10 0 11-5.93-9.14"
                stroke="currentColor"
                fill="none"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              ></path>
              <path
                d="M22 4L12 14.01l-3-3"
                stroke="currentColor"
                fill="none"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              ></path>
            </svg>
            <p class="denhaag-timeline__step-label-text">
              Fourth Step in a Timeline!<svg
                width="1em"
                height="1em"
                viewBox="0 0 24 24"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
                class="MuiSvgIcon-root denhaag-timeline__step-collapse-icon"
                focusable="false"
                aria-hidden="true"
                shape-rendering="auto"
              >
                <path
                  d="M5.293 8.293a1 1 0 011.414 0L12 13.586l5.293-5.293a1 1 0 111.414 1.414l-6 6a1 1 0 01-1.414 0l-6-6a1 1 0 010-1.414z"
                  fill="currentColor"
                ></path>
              </svg>
            </p>
          </div>
          <div
            class="MuiStepContent-root denhaag-timeline__step-content MuiStepContent-last"
            icon="4"
          ></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
